packages/cactus-test-tooling/src/main/typescript/substrate-test-ledger/substrate-test-ledger.ts

```diff
@@ -0,0 +1,202 @@
+  public readonly imageTag: string;
+  public readonly imageFqn: string;
+  public readonly log: Logger;
+  public readonly emitContainerLogs: boolean;
+  public readonly publishAllPorts: boolean;
+  public readonly envVars: Map<string, string>;
+
+  private _containerId: Optional<string>;
+
+  public get containerId(): Optional<string> {
+    return this._containerId;
+  }
+
+  public get container(): Optional<Container> {
+    const docker = new Docker();
+    return this.containerId.isPresent()
+      ? Optional.ofNonNull(docker.getContainer(this.containerId.get()))
+      : Optional.empty();
+  }
+
+  public get className(): string {
+    return SubstrateTestLedger.CLASS_NAME;
+  }
+
+  constructor(public readonly opts: ISubstrateTestLedgerOptions) {
+    const fnTag = `${this.className}#constructor()`;
+    Checks.truthy(opts, `${fnTag} arg options`);
+
+    this.publishAllPorts = opts.publishAllPorts;
+    this._containerId = Optional.empty();
+    this.imageName =
+      opts.imageName || "ghcr.io/hyperledger/cactus-substrate-test-ledger";
+    this.imageTag = opts.imageTag || "0.0.1";
+    this.imageFqn = `${this.imageName}:${this.imageTag}`;
+    this.envVars = opts.envVars || new Map();
+    this.emitContainerLogs = Bools.isBooleanStrict(opts.emitContainerLogs)
+      ? (opts.emitContainerLogs as boolean)
+      : true;
+
+    this.logLevel = opts.logLevel || "INFO";
+
+    const level = this.logLevel;
+    const label = this.className;
+    this.log = LoggerProvider.getOrCreate({ level, label });
+
+    this.log.debug(`Created instance of ${this.className} OK`);
+  }
+  public getContainerImageName(): string {
+    return `${this.imageName}:${this.imageTag}`;
+  }
+  public async start(omitPull = false): Promise<Container> {
+    const docker = new Docker();
+    if (this.containerId.isPresent()) {
+      this.log.warn(`Container ID provided. Will not start new one.`);
+      const container = docker.getContainer(this.containerId.get());
+      return container;
+    }
+    if (!omitPull) {
+      this.log.debug(`Pulling image ${this.imageFqn}...`);
+      await Containers.pullImage(this.imageFqn);
+      this.log.debug(`Pulled image ${this.imageFqn} OK`);
+    }
+
+    const dockerEnvVars: string[] = new Array(...this.envVars).map(
+      (pairs) => `${pairs[0]}=${pairs[1]}`,
+    );
+
+    // TODO: dynamically expose ports for custom port mapping
+    const createOptions = {
+      Env: dockerEnvVars,
+      Healthcheck: {
+        Test: [
+          "CMD-SHELL",
+          `rustup --version && rustc --version && cargo --version`,
+        ],
+        Interval: 1000000000, // 1 second
+        Timeout: 3000000000, // 3 seconds
+        Retries: 10,
+        StartPeriod: 1000000000, // 1 second
+      },
+      ExposedPorts: {
+        "9944/tcp": {}, // OpenSSH Server - TCP
+      },
+      HostConfig: {
+        AutoRemove: true,
+        PublishAllPorts: this.publishAllPorts,
+        Privileged: true,
```

Comment:
@RafaelAPB @CatarinaPedreira Usually we need `Privileged` as `true` for containers that host other containers within, internally. Is this the case with the Substrate AIO image? If there's no specific need for it I'd set it to false to adhere to the principle of least privilege.